Auto Tagging images #29

Sgvkamalakar opened the pull request:
- Implemented image captioning and tag generation functionality using the `Google Gemini AI` service.
- Integrated `Streamlit` for a user-friendly interface.
- Added error handling for invalid API keys and API configuration failures.

This pull request introduces the core functionality for image analysis, enabling users to upload images and generate captions, and tags. The implementation ensures a smooth user experience, validating API keys and handling errors gracefully.

yagyesh-bobde commented:
Please mention the issue in the PR

yagyesh-bobde commented:
And can you create another folder inside this and move your files there.
like 
```
GenAI-Projects/Auto-Image-Tagger/<Your-Unique-Project-Name>/*all your files
```
this will allow other contributors to create more similar projects in the **Auto-Image-Tagger** folder and keep all the image tagging projects separate as well.

yagyesh-bobde commented:
Let me know when all these are done

Sgvkamalakar commented:
## Issue: Auto Tagging images - Chrome Extension #29

### Project Name
Auto Tagging images - Chrome Extension

### Project Description
Given an image input, this tool should generate a set of tags for that image. You can create your models and use them to create a web-based interactive tool or you can even use APIs from Hugging Face or other open-source models.

## What I have contributed:
- Implemented image captioning and tag generation functionality using the `Google Gemini AI` service.
- Integrated `Streamlit` for a user-friendly interface.
- Added error handling for invalid API keys and API configuration failures.

I've revised the PR according to your suggestions by creating a dedicated directory for my contributions, allowing others to add their own contributions in a way that suits them.

This PR introduces the core functionality for image analysis, enabling users to upload images and generate captions, and tags. The implementation ensures a smooth user experience, validating API keys and handling errors gracefully.

## Future work: 
Adding to it,  I am currently developing a Flask-based web application for this.. which enables the creation of an appealable UI.

yagyesh-bobde commented:
Great work @Sgvkamalakar !! 🎉

yagyesh-bobde reviewed (approved):
- [x] Unique Folder Name
- [x] Project description
- [x] GenAI project
- [x] Working
